File: c_stuff/chapter_03/factorial.c
#include <stdio.h>
long double factorial(int num);

int main(void) {
 
  int number;
  long double y;

  printf("Enter a number to find its factorial: ");
  scanf("%i", &number);
  y = factorial(number);
  printf("%i! = %LG\n", number, y);
}

long double factorial(int num){

  long fact; 
  if(num > 1){
   fact = (long)(num * factorial(num-1));
   return fact;
  }
  else
    return 1;
}

File: java_stuff/horton/src/chapter05/Factorial.java
public class Factorial {
	public static void main(String[] args){
		int x = 5;
		System.out.println(x + "! = " +  factorial(x) );
	}

	private static int factorial(int n){
	    return (n==0) ? 1 : n * factorial(n-1);
	}
}
